### PR TITLE
Created Challenge and Submission models

### PR DESCRIPTION
**Pull request by jadenshek**

### CODE REVIEW

File: challenges/submission.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
+from postDB import Model, Column, types
+from datetime import datetime
+
+import utils
+
+
+class Submission(Model):
```
Comment (mohamed040406) [inline challenges/submission.py]: Rename this to `ChallengeSubmission` as we are going to have timathon submissions too

File: challenges/challenge.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    title = Column(types.String)
+    author_id = Column(
+        types.ForeignKey("users", "id", sql_type=types.Integer(big=True)),
+    )
+    description = Column(types.String)
+    rules = Column(types.String)
+    reward = Column(types.Integer)
+
+    def created_at(self) -> datetime:
+        """Returns"""
```
Comment (mohamed040406) [inline challenges/challenge.py]: Returns 👀

File: challenges/submission.py [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+    code = Column(types.String)
+    language = Column(types.String)
+
+    author_id = Column(
+        types.ForeignKey("users", "id", sql_type=types.Integer(big=True)),
+    )
+
+    def created_at(self) -> datetime:
+        """Returns"""
```
Comment (mohamed040406) [inline challenges/submission.py]: Returns 👀

**Review by mohamed040406 — CHANGES_REQUESTED** [review]

**Comment by mohamed040406** [comment]
You should add the models to `models_ordered` in `__init__.py` too

**Comment by jadenshek** [comment]
I changed the class name + said what the property returns

I basically copied from the user model which doesn't specify what created_at returns

File: __init__.py [hunk 4]
```diff
@@ -3,6 +3,8 @@
 
 from .users.user import User
 from .users.token import Token
+from .challenges.challenge import Challenge
```
Comment (SylteA) [inline __init__.py:6]: import from the module, not specific files

File: challenges/challenge.py [hunk 5]
```diff
@@ -0,0 +1,34 @@
+from postDB import Model, Column, types
+from datetime import datetime
+
+import utils
+
+
+class Challenge(Model):
+    """
+    Challenge class to store the challenge details made by specific authors
+    Database Attributes:
```
Comment (SylteA) [inline challenges/challenge.py:10]: missing spacings (check previous models for reference)

> This typically happens when you copy-paste from github :3

**Review by mohamed040406 — APPROVED** [review]
lgtm

File: challenges/challenge.py [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+        Attributes stored in the `challenge` table.
```
Comment (SylteA) [inline challenges/challenge.py:11]: The tablename is `challenges`

File: challenges/challenge_submission.py [hunk 7]
```diff
@@ -0,0 +1,36 @@
+from postDB import Model, Column, types
+from datetime import datetime
+
+import utils
+
+
+class ChallengeSubmission(Model):
+    """
+    Submission class to store the submission details of users for specific challenges
```
Comment (SylteA) [inline challenges/challenge_submission.py:9]: spacings

File: challenges/challenge_submission.py [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+    Database Attributes:
+        Attributes stored in the `submission` table.
```
Comment (SylteA) [inline challenges/challenge_submission.py:11]: tablename is`challengesubmissions`.

File: challenges/challenge_submission.py [hunk 9]
```diff
@@ -0,0 +1,36 @@
...
+    """
+
+    id = Column(types.Integer(big=True), primary_key=True)
+    # Store the ID as a BIGINT even though it's transferred as a string.
+    # This is due to a substantial difference in index time and storage space
+    challenge_id = Column(
+        types.ForeignKey("challenge", "id", sql_type=types.Integer(big=True)),
+    )
+
+    code = Column(types.String)
```
Comment (SylteA) [inline challenges/challenge_submission.py:26]: The CDN has a File model, code should be stored as a file

File: challenges/challenge_submission.py [hunk 10]
```diff
@@ -0,0 +1,36 @@
...
+    language = Column(types.String)
```
Comment (SylteA) [inline challenges/challenge_submission.py:27]: When stored as a file, language can be retrieves from file extension

**Review by SylteA — CHANGES_REQUESTED** [review]

**Pull request merged**